app/src/main/java/com/epfl/drawyourpath/userProfile/UserModel.kt

```diff
@@ -0,0 +1,354 @@
+     * Use this function to modify the email address of the user
+     * @param email new email address
+     */
+    fun setEmailAddress(email: String){
+        if(!checkEMail(email)){
+            throw java.lang.Error("Invalid email format !")
+        }
+        this.emailAddress=email
+    }
+
+    /**
+     * Get the firstname of the user
+     * @return firstname of the user
+     */
+    fun getFirstname(): String{
+        return firstname
+    }
+
+    /**
+     * Get the surname of the user
+     * @return surname of the user
+     */
+    fun getSurname(): String{
+        return surname
+    }
+
+    /**
+     * Get the date of birth of the user
+     * @return date of birth of the user
+     */
+    fun getDateOfBirth(): LocalDate{
+        return dateOfBirth
+    }
+
+    /**
+     * Get the daily distance goal of the user
+     * @return daily distance goal of the user
+     */
+    fun getDistanceGoal(): Double{
+        return distanceGoal
+    }
+
+    /**
+     * Use this function to modify the daily distance goal of the user
+     * @param distance new daily distance goal
+     */
+    fun setDistanceGoal(distance: Double){
+        if(distance == 0.0){
+            throw java.lang.Error("The distance goal can't be equal to 0 !")
+        }
+        this.distanceGoal=distance
+    }
+
+    /**
+     * Get the daily activity time goal of the user
+     * @return daily activity time goal of the user
+     */
+    fun getActivityTime(): Double{
+        return activityTimeGoal
+    }
+
+    /**
+     * Use this function to modify the daily activity time goal of the user
+     * @param time new daily activity time goal
+     */
+    fun setActivityTimeGoal(time: Double){
+        if(time == 0.0){
+            throw java.lang.Error("The activity time goal can't be equal to 0 !")
+        }
+        this.activityTimeGoal=time
+    }
+
+    /**
+     * Get the daily number of paths goal of the user
+     * @return daily number of paths goal of the user
+     */
+    fun getNumberOfPathsGoal(): Int{
+        return nbOfPathsGoal
+    }
+
+    /**
+     * Use this function to modify the daily number of paths goal of the user
+     * @param nbOfPaths new daily number of paths goal
+     */
+    fun setNumberOfPathsGoal(nbOfPaths: Int){
+        if(nbOfPaths == 0){
+            throw java.lang.Error("The number of paths goal can't be equal to 0 !")
+        }
+        this.nbOfPathsGoal=nbOfPaths
+    }
+
+    /**
+     * Get the age of the user
+     * @return the age of the user
+     */
+    fun getAge(): Int{
+        var age = LocalDate.now().year-dateOfBirth.year
+        if(LocalDate.now().minusYears(age.toLong())<=dateOfBirth){
+            age -= 1
+        }
+        return age
+    }
+
+    /**
+     * This function will remove the user with username to the friend list
+     * @param username of the user that we want to remove
+     */
+    fun removeFriend(username: String) {
+        if(!friendsList.contains(username)) {
```

Comment:
Should update the friends in the database. (Maybe add a TODO as it is in a following PR).